inventory-repository/src/main/scala/com/normation/inventory/ldap/core/InventoryMapper.scala

```diff
@@ -775,10 +775,63 @@ class InventoryMapper(
     }
   }
 
-    def nodeFromEntry(entry:LDAPEntry) : Box[NodeInventory] = {
-    def missingAttr(attr:String) : String = "Missing required attribute %s".format(attr)
+  def mapOsDetailsFromEntry(entry: LDAPEntry) : Box[OsDetails] = {
     def requiredAttr(attr:String) = entry(attr) ?~! missingAttr(attr)
+    for {
+      osName        <- requiredAttr(A_OS_NAME)
+      osVersion     <- requiredAttr(A_OS_VERSION).map(x => new Version(x))
+      kernelVersion <- requiredAttr(A_OS_KERNEL_VERSION).map(x => new Version(x))
+      osFullName    =  entry(A_OS_FULL_NAME).getOrElse("")
+      osServicePack =  entry(A_OS_SERVICE_PACK)
+      osDetails     <- if(entry.isA(OC_WINDOWS_NODE)) {
+                          val os = osName match {
+                            case A_OS_WIN_XP => WindowsXP
+                            case A_OS_WIN_VISTA => WindowsVista
+                            case A_OS_WIN_SEVEN => WindowsSeven
+                            case A_OS_WIN_2000 => Windows2000
+                            case A_OS_WIN_2003 => Windows2003
+                            case A_OS_WIN_2008 => Windows2008
+                            case A_OS_WIN_2008_R2 => Windows2008R2
+                            case A_OS_WIN_2012 => Windows2012
+                            case _ => UnknownWindowsType
+                          }
+                          val userDomain = entry(A_WIN_USER_DOMAIN)
+                          val registrationCompany = entry(A_WIN_COMPANY)
+                          val productKey = entry(A_WIN_KEY)
+                          val productId = entry(A_WIN_ID)
+                          Full(Windows(os,osFullName,osVersion,osServicePack,kernelVersion,userDomain,registrationCompany,productKey,productId))
+
+                        } else if(entry.isA(OC_LINUX_NODE)) {
+                          val os = osName match {
+                            case A_OS_DEBIAN  => Debian
+                            case A_OS_UBUNTU  => Ubuntu
+                            case A_OS_REDHAT  => Redhat
+                            case A_OS_CENTOS  => Centos
+                            case A_OS_FEDORA  => Fedora
+                            case A_OS_SUZE    => Suse
+                            case A_OS_ORACLE  => Oracle
+                            case A_OS_SCIENTIFIC => Scientific
+                            case A_OS_ANDROID => Android
+                            case _            => UnknownLinuxType
```

Comment:
That's not at all the goal of that bug fixe, whose will is just to make the method visible, and add efficient queries for software by node. But why not on the general arch evolution.